tools/autograd/derivatives.yaml

```diff
@@ -1932,6 +1932,16 @@
   self: at::view_as_real(grad.contiguous().resolve_conj()) # [gx, gy]
   result: at::view_as_complex(self_t)
 
+- name: view_as(Tensor(a) self, Tensor other) -> Tensor(a)
+  dispatch:
+    Default:
+      # the default case will use the CompositeImplicitAutograd impl
+      self: not_implemented("view_as")
+      other: non_differentiable
+    AutogradNestedTensor:
+      self: grad.view_as(self)
```

Comment:
This will safe the full self? You most likely want only .size() or some lightweight thing here?

Reply:
yep that's right, sadly. this PR is an inefficient workaround for our current lack of factory function support with shapes that have nested ints. I'm not sure of another way to address this without that support